File: FbxCppTests/LayerTest.cpp

#include "Tests.h"

using namespace std;

void Layer_SetVisibility_SetsVisibility()
{
    // given:
    FbxManager* manager = FbxManager::Create();
    FbxLayerContainer* lc = FbxLayerContainer::Create(manager, "");
    lc->CreateLayer();
    FbxLayer* layer = lc->GetLayer(0);
    FbxLayerElementVisibility* lev = FbxLayerElementVisibility::Create(manager, "");

    // require:
    AssertEqual(NULL, layer->GetVisibility());

    // when:
    layer->SetVisibility(lev);

    // then:
    AssertEqual(lev, layer->GetVisibility());
}

void Layer_SetVisibility_ReplacesPreviousVisibilityElement()
{
    // given:
    FbxManager* manager = FbxManager::Create();
    FbxLayerContainer* lc = FbxLayerContainer::Create(manager, "");
    lc->CreateLayer();
    FbxLayer* layer = lc->GetLayer(0);
    FbxLayerElementVisibility* lev1 = FbxLayerElementVisibility::Create(manager, "one");
    FbxLayerElementVisibility* lev2 = FbxLayerElementVisibility::Create(manager, "two");
    layer->SetVisibility(lev1);;

    // require:
    AssertEqual(lev1, layer->GetVisibility());

    // when:
    layer->SetVisibility(lev2);

    // then:
    AssertEqual(lev2, layer->GetVisibility());
}

void LayerTest::RegisterTestCases()
{
    AddTestCase(Layer_SetVisibility_SetsVisibility);
    AddTestCase(Layer_SetVisibility_ReplacesPreviousVisibilityElement);
}

File: FbxSharpTests/LayerTest.cs
using System;
using NUnit.Framework;
using FbxSharp;

namespace FbxSharpTests
{
    [TestFixture]
    public class LayerTest : TestBase
    {
        [Test]
        public void Layer_SetVisibility_SetsVisibility()
        {
            // given:
            var lc = new FbxMesh("");
            lc.CreateLayer();
            var layer = lc.GetLayer(0);
            var lev = new FbxLayerElementVisibility("");

            // require:
            Assert.AreEqual(null, layer.GetVisibility());

            // when:
            layer.SetVisibility(lev);

            // then:
            Assert.AreEqual(lev, layer.GetVisibility());
        }

        [Test]
        public void Layer_SetVisibility_ReplacesPreviousVisibilityElement()
        {
            // given:
            var lc = new FbxMesh("");
            lc.CreateLayer();
            var layer = lc.GetLayer(0);
            var lev1 = new FbxLayerElementVisibility("one");
            var lev2 = new FbxLayerElementVisibility("two");
            layer.SetVisibility(lev1);;

            // require:
            Assert.AreEqual(lev1, layer.GetVisibility());

            // when:
            layer.SetVisibility(lev2);

            // then:
            Assert.AreEqual(lev2, layer.GetVisibility());
        }
    }
}
